src/Stratis.Bitcoin.Features.BlockStore/BlockStoreSignaled.cs

```diff
@@ -226,7 +226,8 @@ protected override void OnNextCore(Block block)
             }
 
             // Announce the blocks to each of the peers.
-            IEnumerable<BlockStoreBehavior> behaviours = peers.Select(s => s.Behavior<BlockStoreBehavior>());
+            IEnumerable<BlockStoreBehavior> behaviours = peers.Select(s => s.Behavior<BlockStoreBehavior>())
```

Comment:
Can the behavior be null here?

Reply:
Those null behaviors- are they added from the peer discovery? If so- don't add connectionManagerBehavior during peer creation in discovery

Reply:
OK I'll check in more details, the issue I was seeing was that upon Disposing of the node, there was still some batches left to send, and then this causes an exception as the behavior is being disposed so it ends up in a strange state.
I think this change is correct anyway, but we could do try and avoid these situation passing and handling correctly a cancellation token...
I'll check in more details what you were advising, thanks